static/js/ducks/instrument.js

```diff
@@ -0,0 +1,44 @@
+import messageHandler from "baselayer/MessageHandler";
+
+import * as API from "../API";
+import store from "../store";
+
+const REFRESH_INSTRUMENT = "skyportal/REFRESH_INSTRUMENT";
+
+const FETCH_INSTRUMENT = "skyportal/FETCH_INSTRUMENT";
+const FETCH_INSTRUMENT_OK = "skyportal/FETCH_INSTRUMENT_OK";
+
+const SUBMIT_INSTRUMENT = "skyportal/SUBMIT_INSTRUMENT";
+
+export const fetchInstrument = (id) =>
+  API.GET(`/api/instrument/${id}`, FETCH_INSTRUMENT);
+
+export const submitInstrument = (run) =>
+  API.POST(`/api/instrument`, SUBMIT_INSTRUMENT, run);
+
+// Websocket message handler
+messageHandler.add((actionType, payload, dispatch, getState) => {
+  const { instrument } = getState();
+  if (actionType === REFRESH_INSTRUMENT) {
+    const { instrument_id } = payload;
```

Comment:
Now I explicitly call instruments to get the new list. That ok?

Reply:
That seems to work when adding from the front end. This is probably because you explicitly call `fetchInstruments` from the dispatch. If someone used the API to post a new instrument it would not show up unless you refresh. 
The way around this is to send a REFRESH_INSTRUMENTS websocket message from the handler (instrument.py) and have that picked up by the reducer (instruments.js). Look at what is done in groups (group.py and groups.js).